src/averageAges.js

```diff
@@ -1,61 +1,34 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = (century)
+    ? people.filter(person => person.sex === 'm'
+       && (Math.ceil(person.died / 100) === century))
+    : people.filter(person => person.sex === 'm');
+
+  return men.reduce((prev, current) =>
+    prev + current.died - current.born, 0) / men.length;
```

Comment:
You have a repetitive operation, would be good practice to create a function which will take an array and return result